NLog.Windows.Forms.Tests/RichTextBoxTargetTests.cs

```diff
@@ -588,13 +582,234 @@ public void ManualRegisterTestWithRetention()
                 Assert.Same(form, target.TargetForm);
                 Assert.Same(rtb, target.TargetRichTextBox);
 
-                MemoryStream ms = new MemoryStream();
-                target.TargetRichTextBox.SaveFile(ms, RichTextBoxStreamType.RichText);
-                string result = Encoding.UTF8.GetString(ms.GetBuffer());
+                string result = ExtractRtf(target.TargetRichTextBox);
 
                 Assert.True(result.Contains(@"No Control"));
                 Assert.True(result.Contains(@"Has Control"));
             }
         }
+
+        [Fact]
+        public void LinkLayoutTestDisabledLinks()
+        {
+            RichTextBoxTarget target = new RichTextBoxTarget()
+            {
+                ControlName = "Control1",
+                UseDefaultRowColoringRules = true,
+                Layout = "${level} ${logger} ${message} ${rtb-link:inner=descr}",
+                ToolWindow = false,
+                Width = 300,
+                Height = 200,
+                SupportLinks = false
+            };
+
+            SimpleConfigurator.ConfigureForTargetLogging(target, LogLevel.Trace);
+            logger.Info("Test");
+
+            Application.DoEvents();
+
+            string result = ExtractRtf(target.TargetRichTextBox);
+            Assert.True(Regex.IsMatch(result, @"(\([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}\))"));  //the placeholder GUID was not replaced by was not replaced because of SupportLinks set to false
+        }
+
+        [Fact]
+        public void LinkTest()
+        {
+            RichTextBoxTarget target = new RichTextBoxTarget()
+            {
+                ControlName = "Control1",
+                UseDefaultRowColoringRules = true,
+                Layout = "${level} ${logger} ${message} ${rtb-link:inner=descr}",
+                ToolWindow = false,
+                Width = 300,
+                Height = 200,
+                SupportLinks = true
+            };
+
+            SimpleConfigurator.ConfigureForTargetLogging(target, LogLevel.Trace);
+            logger.Info("Test");
+
+            Application.DoEvents();
+
+            Assert.Same(target, RichTextBoxTarget.GetTargetByControl(target.TargetRichTextBox));
+
+            string resultRtf = ExtractRtf(target.TargetRichTextBox);
+            string resultText = target.TargetRichTextBox.Text;
+            Assert.False(Regex.IsMatch(resultRtf, @"(\([a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}\))"));  //the placeholder GUID was replaced
+            //cant check for specific ids here because of the possible parallel execution
```

Comment:
Actually, I'm not sure that this is because of parallel execution. It could be just that executor is not reloading classes, so static fields are not re-initialized. Anyway I was able to test all that I thought was necessary.